framework/src/meshgenerators/ElementGenerator.C

```diff
@@ -37,6 +37,8 @@ ElementGenerator::validParams()
 
   params.addClassDescription("Generates individual elements given a list of nodal positions.");
 
+  // Declare that this generator has a generateData method
+  MeshGenerator::setHasGenerateData(params);
```

Comment:
Why did you need to make this one support `generateData()`? It doesn't do anything in data only mode, and I think that's not clear here to the user

Reply:
I did this to add a test that could be used by the Reactor test app as a proxy for that last generator that would be used in place of the Griffin generator. I guess now that AddMetaDataGenerator has a `setHasGenerateData` method, I should use this in my test instead of ElementGenerator?

Reply:
Actually `AddMetaDataGenerator` wouldn't work for the data driven test in the Reactor module since it doesn't actually generate an output mesh from scratch. The reasoning for adding a data only method `ElementGenerator` was so that it could retain the metadata defined from Reactor module and still generate an output mesh by setting it as the last generator in the generator workflow.

Reply:
ok this makes sense.
The way to do this however is not to modify the ElementMG, but to create a test object in the reactor module. it can inherit from any MG that suits your needed, and is register to the ReactorTestApp. Then you can use that test object for this purpose

Reply:
I understand what you’re saying, though I’ve never created a test mesh generator before. Is this just a matter of defining a new MeshGenerator in modules/reactor/test/src/meshgenerators?

Reply:
Yup exactly

Reply:
Yeah, please do that if you can. I think that it's pretty dangerous to let this support data only mode when it doesn't accomplish anything. We should limit data only to objects that we have tight control over in reactor

Reply:
I agree this is better design for testing. I've gone ahead and reverted the changes to ElementGenerator and created a custom test mesh generator to accomplish this task